### ISSUE TITLE
genericExtensions breaks code on macOS but works on iOS

### ISSUE THREAD

**Issue by honghaoz** [nicklockwood/SwiftFormat#1276]
- swiftformat version: `0.50.0`
- Xcode version: `Version 14.0 (14A309)`
- Swift version: `swift-driver version: 1.62.8 Apple Swift version 5.7 (swiftlang-5.7.0.127.4 clang-1400.0.29.50)`
- macOS version: `macOS Monterey 12.5.1 (21G83)`

The original code is:
```swift
public extension Collection where Element == Int {

  func sum() -> Element {
    reduce(0, +)
  }
}
```

The formatted code is:
```swift
public extension Collection<Int> {

  func sum() -> Element {
    reduce(0, +)
  }
}
```

The code is from a SPM package, building for iOS targets works well.
However building for macOS targets shows errors:
```
Cannot specialize protocol type 'Collection'
```

Is this expected? 

Might be related: https://forums.swift.org/t/xcode-14-rc-cannot-specialize-protocol-type/60171

**Comment by calda** [comment]
This is because macOS 12 doesn't support the full set of features in Swift 5.7, which is required to use this rule. 

Are you specifying `--swiftversion 5.7`? I'd suggest using `--swiftversion 5.6` until this issue with the macOS SDK is resolved. Another workaround would be to manually disable the `genericExtensions` rule in your config file.

**Comment by honghaoz** [comment]
@calda yeah, make sense! I used 5.7, specified in a `.swift-version` file.
Right now I used `// swiftformat:disable:next genericExtensions` to temporarily disable the rule.

I also found that Array works fine. For example:
```swift
public extension [Int] {
  func hey() {}
}

public extension Array<Int> {
  func hey() {}
}
```
